tests/components/mqtt/test_common.py

```diff
@@ -473,3 +475,47 @@
     assert state is not None
     for topic in topics:
         mock_mqtt.async_subscribe.assert_any_call(topic, ANY, ANY, ANY)
+
+
+async def help_test_entity_id_update_discovery_update(
+    hass, mqtt_mock, domain, config, topic=None
+):
+    """Test MQTT discovery update after entity_id is updated."""
+    # Add unique_id to config
+    config = copy.deepcopy(config)
+    config[domain]["unique_id"] = "TOTALLY_UNIQUE"
+
+    if topic is None:
+        # Add default topic to config
+        config[domain]["availability_topic"] = "avty-topic"
+        topic = "avty-topic"
+
+    entry = MockConfigEntry(domain=mqtt.DOMAIN)
+    entry.add_to_hass(hass)
+    await async_start(hass, "homeassistant", {}, entry)
+    ent_registry = mock_registry(hass, {})
+
+    data = json.dumps(config[domain])
+    async_fire_mqtt_message(hass, f"homeassistant/{domain}/bla/config", data)
+    await hass.async_block_till_done()
+
+    async_fire_mqtt_message(hass, topic, "online")
+    state = hass.states.get(f"{domain}.test")
+    assert state.state != STATE_UNAVAILABLE
+
+    async_fire_mqtt_message(hass, topic, "offline")
+    state = hass.states.get(f"{domain}.test")
+    assert state.state == STATE_UNAVAILABLE
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5273824)